components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -1421,4 +1431,36 @@ private void setLoggedInOrgIdInRequest(AuthenticatedIdPData authenticatedIdPData
             }
         }
     }
+
+    /**
+     * Resolve user resident organization for the users authenticated via the B2B organization SSO authenticator.
+     *
+     * @param authenticatedUser   The authenticated user.
+     * @param authenticatorConfig The authenticator configurations.
+     * @return The organization where the user's identity is managed.
+     */
+    private String resolveUserResidentOrganization(AuthenticatedUser authenticatedUser,
+                                                   AuthenticatorConfig authenticatorConfig) {
+
+        // Check for user organization claim for the authenticated user via the organization SSO authenticator.
+        if (authenticatedUser.getUserAttributes() != null && isLoggedInWithOrganizationSSO(authenticatorConfig)) {
+            for (Map.Entry<ClaimMapping, String> userAttributes : authenticatedUser.getUserAttributes().entrySet()) {
+                if (FrameworkConstants.USER_ORGANIZATION_CLAIM.equals(
+                        userAttributes.getKey().getLocalClaim().getClaimUri())) {
+                    return userAttributes.getValue();
+                }
+            }
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String resolveTenantDomain(String organizationId) throws FrameworkException {
```

Comment:
If we are going to add the method description to private methods, shall we add that to here as well?

Reply:
yeah we can, but for private methods, the method description is not required unless its logic is hard to understand. But this method is simple. The above private method need more context info to understand.